engine/grid.py

```diff
@@ -202,40 +222,32 @@ def get_neighbor_node(self, row, col, row_max, col_max):
         else:
             return neighbor_node
 
-
+    ##Activates rectangle based on row start/end, col start/end
     def activate_rectangle(self, row, col, row_limit, col_limit):
-
         """
         Activates all the nodes in a rectangle based on row start/end, col
         start/end.
         """
         for x in range(row, row_limit):
             for y in range(col, col_limit):
-                self.nodes[x,y].is_active = True
+                self.nodes[x, y].is_active = True
         return self.nodes
 
-
-    
+    ##Activates circle based on center and radius
     def activate_circle(self, circle_center_row, circle_center_col, circle_radius):
         """
         Activates all the nodes in a circle based on center and radius.
         """
-        rows = self.nodes.shape[0]
-        cols = self.nodes.shape[0]
-        for x in range(rows):
-            for y in range(cols):
-                if ((x-circle_center_row)**2 + (y-circle_center_col)**2 - circle_radius**2) < 0:
+        for x in range(self.num_rows):
+            for y in range(self.num_cols):
+                if ((x - circle_center_row) ** 2 + (y - circle_center_col) ** 2 - circle_radius ** 2) < 0:
                     self.nodes[x, y].is_active = True
 
-   
-
-    def is_inside_triangle(self, x1, y1, x2, y2, x3, y3, x, y):
-        """
-        A function to check whether point P(x, y) lies inside the triangle 
-        formed by A(x1, y1), B(x2, y2) and C(x3, y3)
-        """
+    # A function to check whether point P(x, y)
```

Comment:
I think I actually prefer this as a spec than a comment. Should also specify the function param (description, type), and return (description, type)

Reply:
<img width="696" alt="Screen Shot 2023-03-09 at 4 26 29 PM" src="https://user-images.githubusercontent.com/56476240/224163518-8d35da4c-d0df-4e7f-84a6-8ae99917af9b.png">